src/components/Mapviewer.tsx

```diff
@@ -1,161 +1,129 @@
-            source={'measuringstation'}
-            source-layer={'default'}
-            layout={{
-              "text-field": "{object_name}",
-              "text-size": 14,
-              "text-anchor": "bottom-left",
-              "icon-image": "pumpIconImage",
-              "icon-anchor": "bottom-right",
-              "icon-size": 0.1
+        <TileLayer
+          url={`https://api.mapbox.com/styles/v1/nelenschuurmans/ck8sgpk8h25ql1io2ccnueuj6/tiles/256/{z}/{x}/{y}@2x?access_token=${mapBoxAccesToken}`}
+        />
+        {timestamps.map(timestamp => (
+          <WMSTileLayer
+            key={timestamp.time}
+            url={'/wms/'}
+            layers={'radar:5min'}
+            styles={'radar-5min'}
+            time={timestamp.time}
+            format={'image/png'}
+            uppercase={true}
+            bounds={getBounds(raster)}
+            tileSize={1024}
+            opacity={timestamp.time === timestamps[step].time ? 1 : 0}
+            onload={() => {
+              console.log('finish loading tiles for ', timestamp.time)
+              setTimestamps(timestamps.map(ts => {
+                if (ts.time === timestamp.time) {
+                  return {
+                    ...ts,
+                    loaded: true
+                  }
+                };
+                return ts;
+              }))
             }}
-            paint={{
-              // "circle-radius": 4,
-              // "circle-stroke-width": 1,
-              // "circle-stroke-color": 'grey',
-              // "circle-color": [
-              //   'case',
-              //   ['>', ["get", "object_id"], 1000],
-              //   'red',
-              //   'blue'
-              // ],
-              "text-color": [
-                'case',
-                ['>', ["get", "object_id"], 1000],
-                'blue',
-                'red'
-              ],
-              "icon-color": [
-                'case',
-                ['>', ["get", "object_id"], 1000],
-                'blue',
-                'red'
-              ],
-              // "icon-color": [
-              //   'match',
-              //   ["get", "object_name"],
-              //   "ZWOLLE",
-              //   'red',
-              //   "Dante",
-              //   'brown',
-              //   'blue'
-              // ]
+            // the onTileLoadStart happens when a tile is requested and starts loading
+            // in this case, stop the animation and set the "loaded" parameter back to false
+            // to indicate that the tiles are being reloaded
+            // However, it normally takes very long to reload all the tiles (way longer than the first time)
+            // During this time, user is not supposed to zoom in/zoom out, else the reloading process will happen again
+            ontileloadstart={() => {
+              stopAnimation()
+              setTimestamps(timestamps.map(ts => ({
+                ...ts,
+                loaded: false
```

Comment:
This seems like a problem, it sets _all_ the loaded things to false, for all timestamps? And each timestamp has its own ontileloadstart?

Should probably pass a function to setTimestamp, that takes the current value and then only sets the right one, to prevent race conditions later.

Reply:
Also I'm not sure it's acceptable to stop animating whenever the user pans or zooms.

Reply:
You are right. It could introduce a race conditions. I fixed it. 

For zooming and panning issue, it could pause to reload instead of stop and then resume.